app/src/main/java/com/example/mvvmcleanarchitecture/core/di/AppModule.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.mvvmcleanarchitecture.core.di
+
+import android.content.Context
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import org.koin.android.ext.koin.androidContext
+import org.koin.dsl.module
+
+val appModule = module {
+    factory { provideLinearLayoutManager(androidContext()) }
+
+    factory { provideGridLayoutManager(androidContext()) }
+
+    factory { provideDividerItemDecoration(androidContext()) }
```

Comment:
tutaj nie byłem pewien, czy zamysł był taki że to jest singleton i za każdym razem gdy chcemy dokonfigurować nasz recyclerView to korzystając z 1 beana na cały app ? 
I ponawiam pytanie: dlaczego takiej konfiguracji nie robimy w xml ? Ze względu na to by nie tracić możliwości reużywalności takiego recyclerViewera w różnych miejscach ?

Reply:
Mnie osobiście bardziej przekonuje w kontekście komponentów ui podeście tworzenia ich jako factory. W innym przypadku nagle by się okazało, że całą apkę mamy ojechaną na singlach :) Co do konfiguracji xml vs moduł, to w zasadzie za modułem przemawia jedna kwestia - łatwość modyfikacji i elastyczność. Zmieniasz w jednym miejscu i propaguje się tam gdzie jest użyte. To jest właśnie zaleta inversion of control. Chociaż w przypadku separatora to może nie mieć aż takiego znaczenia i również można tego użyć w xml